### PR TITLE
feat(platform): support forgejo

### PR DESCRIPTION
**Pull request by viceice**
## Changes
- support Forgejo via Gitea platform
- fix some platform type issues (should i separate that?)

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Review by rarkins — CHANGES_REQUESTED** [review]
I think best to separate out the type fixing. Also are you sure it's clear enough to Forgejo users to simply set platform=gitea and use Gitea documentation?

**Comment by viceice** [comment]
> I think best to separate out the type fixing. 

ok

> Also are you sure it's clear enough to Forgejo users to simply set platform=gitea and use Gitea documentation?

Yes, that's what the Forgejo comunitxy currently also suggests.
I'll be much more work to instroduce a new forgejo platform.
Would require a lot refactorings for gitea platform to replace the host type.
I would prefer to reuse gitea platform until they will change in incompatible ways.

File: lib/modules/platform/gitea/index.md [hunk 1]
```diff
@@ -1,8 +1,7 @@
-# Gitea
+# Gitea and Forgejo
 
-Renovate uses modern Git upload filters to suppress large blob downloads.
-For Gitea you need to manually enable upload filters.
-Read the official [Gitea docs](https://docs.gitea.io/en-us/clone-filters/) for more information.
+Renovate supports [Gitea](https://gitea.io) and the new fork [Forgejo](https://forgejo.org).
+Forgejo exists since `v1.18.0` and is currently fully compatible to Gitea.
```
Comment (HonkingGoose) [inline lib/modules/platform/gitea/index.md:4]: Do we really need to mention the Forgejo version number? I imagine the important bit is that Forgejo is fully compatible with Gitea?

Comment (viceice) [reply lib/modules/platform/gitea/index.md:4]: 🤷‍♂️ currently the version number of forgejo follows the Gitea numbers.

**Comment by rarkins** [comment]
Could we support&massage platform=forgejo to platform=gitea? Or better to keep it simple?

**Comment by viceice** [comment]
> Could we support&massage platform=forgejo to platform=gitea? Or better to keep it simple?

Would like to keep it simple for now.

**Review by rarkins — APPROVED** [review]

**Review by rarkins — APPROVED** [review]

**Review by rarkins — APPROVED** [review]

**Pull request merged**